backend/src/apportionment/mod.rs

```diff
@@ -1,199 +1,350 @@
+            let pg_number = get_pg_number_with_largest_average(
+                pg_votes,
+                whole_seats,
+                &rest_seats,
+                remaining_seats,
+            )?;
+
+            *rest_seats.entry(pg_number).or_insert(0) += 1;
+            remaining_seats -= 1;
+            // pg_number_last_remaining_seat = Some(pg_number);
+            info!(
+                "Remaining seat assigned using greatest average system to pg_number: {}",
+                pg_number
+            );
+        }
+    } else {
+        info!("Remaining seats calculation for less than 19 seats.");
+        // using greatest surpluses system ("stelsel grootste overschotten")
+        let mut surpluses =
+            get_surplus_per_pg_where_total_votes_meets_the_threshold(pg_votes, whole_seats, quota);
+        while remaining_seats > 0 {
+            info!("======================================================");
+            debug!("Remaining seats: {}", remaining_seats);
+            if !surpluses.is_empty() {
+                // assign remaining seat to the party with the largest surplus and
+                // remove that party and surplus from the list
+                let pg_number = get_pg_number_with_largest_surplus(&surpluses, remaining_seats)?;
+
+                *rest_seats.entry(pg_number).or_insert(0) += 1;
+                surpluses.remove(&pg_number);
+                remaining_seats -= 1;
+                // pg_number_last_remaining_seat = Some(pg_number);
+                info!(
+                    "Remaining seat assigned using greatest surplus system to pg_number: {}",
+                    pg_number
+                );
+            } else {
+                // once there are no parties with surpluses left and more remaining seats exist,
+                // assign remaining seat to the party with the largest average
+                // using greatest averages system ("stelsel grootste gemiddelden")
+                let pg_number = get_pg_number_with_largest_average(
+                    pg_votes,
+                    whole_seats,
+                    &rest_seats,
+                    remaining_seats,
+                )?;
+                // TODO: The formal description states that initially rest seats can only be
+                //  assigned to unique parties, but with even more remaining seats then multiple
+                //  seats can be assigned to the same party.
+                //  Should there be any limitation built in for this theoretical case?
+                *rest_seats.entry(pg_number).or_insert(0) += 1;
+                remaining_seats -= 1;
+                // pg_number_last_remaining_seat = Some(pg_number);
+                info!(
+                    "Remaining seat assigned using greatest average system to pg_number: {}",
+                    pg_number
+                );
+            }
+        }
+    }
+    Ok(rest_seats)
+}
+
 /// Apportionment - work in progress!!
```

Comment:
```suggestion
/// Apportionment
```

I'd say the whole application is work in progress, so no need to say so explicitly :). At least add a TODO for discoverability